templates/user_profile.html

```diff
@@ -14,213 +14,187 @@
-                </div>
-                <form class="form" action="{{ url_for('user.update_user_profile', user_id=requested_user.id) }}" method="post"
-                      id="profileForm">
-                    <div class="row">
-                        <div class="col-sm-6 col-md-6">
-                            <dl>
-                                <dt>First name</dt>
-                                <dd>
-                                    <input type="text" class="form-control" name="first_name" id="first_name"
-                                           placeholder="first name" value="{{ requested_user.first_name }}">
-                                </dd>
-                                <dt>Last name</dt>
-                                <dd>
-                                    <input type="text" class="form-control" name="name" id="name"
-                                           placeholder="last name" value="{{ requested_user.name }}">
-                                </dd>
-                                <dt>Email</dt>
+                    <div class="col-sm-6 col-md-6">
+                        <dl>
+                            <dt>Status</dt>
+                            <dd>
+                                <div class="form-check">
+                                    <input type="checkbox" class="form-check-input" id="is_teacher"
+                                           name="is_teacher"
+                                           {% if requested_user.is_teacher %}checked{% endif %}
+                                            {% if current_user and not requested_user.admin %}
+                                           disabled {% endif %}>
+                                    <label class="form-check-label" for="is_teacher">Teacher</label>
+                                </div>
+                                <div class="form-check">
+                                    <input type="checkbox" class="form-check-input" id="is_researcher"
+                                           name="is_researcher"
+                                           {% if requested_user.is_researcher %}checked{% endif %}
+                                            {% if current_user and not requested_user.admin %} disabled {% endif %}>
+                                    <label class="form-check-label" for="is_researcher">Researcher</label>
+                                </div>
+                            </dd>
+                            <div class="form-group" id="supervisor_select" style="display: none;">
+                                <dt>Supervisor</dt>
                                 <dd>
-                                    <input type="email" class="form-control" name="email" id="email"
-                                           value="{{ requested_user.email }}" {% if current_user %}
-                                           readonly {% endif %}>
+                                    {% with page='profile' %}
```

Comment:
Conditions :

- supervisor select disabled if `current_user and not requested_user.admin`
- researcher defined => show supervisor list